Cabal/tests/UnitTests/Distribution/Types/GenericPackageDescription.hs

```diff
@@ -35,3 +41,33 @@ throwsUndefined :: NFData a => String -> a -> Assertion
 throwsUndefined field a =
   C.catch (C.evaluate (rnf a) >> assertFailure ("Deepseq failed to evaluate " ++ show field))
           (\(C.ErrorCall _) -> return ())
+
+----- Verify duplicate flag recognition in "FlagAssignment" -------------------
+
+testFindDuplicateFlagAssignments :: Assertion
+testFindDuplicateFlagAssignments = do
+  let flag n v = (mkFlagName n, v)
+      findDuplicates = findDuplicateFlagAssignments . mkFlagAssignment
+  forM_ [ (b1,b2) | b1 <- [True,False], b2 <- [True,False] ] $ \(v1,v2) -> do
+    assertEqual "" [] (findDuplicates [flag "foo" v1, flag "bar" v2])
+    assertUnequal "" [] (findDuplicates [flag "foo" v1, flag "FOO" v2])
+
+assertUnequal :: (HasCallStack, Eq a, Show a) => String -> a -> a -> Assertion
+assertUnequal preface expected actual =
+  when (actual == expected) (assertFailure msg)
+    where msg = (if null preface then "" else preface ++ "\n") ++
+                "unexpected: " ++ show actual
+
+genFlagName :: Gen FlagName
+genFlagName = mkFlagName <$> listOf1 (elements (['a'..'z'] ++ ['A'..'Z']))
+
+genFlagValue :: Gen (FlagName, Bool)
+genFlagValue = (,) <$> genFlagName <*> arbitrary
+
+propertyFlagNamesAreCaseInsensitive :: Property
+propertyFlagNamesAreCaseInsensitive =
+  forAll (listOf1 (resize 2 genFlagValue)) $ \fa ->
+    let fa'        = nub [ map toLower (unFlagName n) | (n,_) <- fa ]
+        duplicates = findDuplicateFlagAssignments (mkFlagAssignment fa)
+    in
+       length duplicates == length fa - length fa'
```

Comment:
use `===` (triple equals) to print LHS and RHS

Reply:
@phadej, I force-pushed a new version (72e4f5575243305b3524d5666f5660c3f965b9c3) that uses `===`.